src/misc_gui.cpp

```diff
@@ -961,21 +975,63 @@ struct QueryStringWindow : public Window
 			case WID_QS_CANCEL:
 				this->Close();
 				break;
+
+			case WID_QS_MOVE:
+				this->last_user_action = widget;
+
+				if (Station::IsExpected(Station::Get(this->parent->window_number))) { // this is a station
+					SetViewportStationRect(Station::Get(this->parent->window_number), !this->IsWidgetLowered(WID_QS_MOVE));
+				} else { //this is a waypoint
+					SetViewportWaypointRect(Waypoint::Get(this->parent->window_number), !this->IsWidgetLowered(WID_QS_MOVE));
+				}
```

Comment:
I don't love the comments after an opening brace (and tend not to use inline comments anyway in favor of the suggested style below).

Suggestion also applies to `OnPlaceObject()` and `OnPlaceObjectAbort()`.

```suggestion
				if (Station::IsExpected(Station::Get(this->parent->window_number))) {
					/* This is a station. */
					SetViewportStationRect(Station::Get(this->parent->window_number), !this->IsWidgetLowered(WID_QS_MOVE));
				} else {
					/* This is a waypoint. */
					SetViewportWaypointRect(Waypoint::Get(this->parent->window_number), !this->IsWidgetLowered(WID_QS_MOVE));
				}
```